### PR TITLE
Add `load_files` and deprecate `load_file`

### PR DESCRIPTION
**Pull request by leina05**
PR checklist:

- [ ] (maybe) improve the way we do warnings in Python so that it includes location of source of warning
- [ ] update documentation
- [ ] Added changelog entry.

### CODE REVIEW

File: polar-core/src/polar.rs [hunk 1]
```diff
@@ -238,6 +234,15 @@ impl Polar {
...
     }
 
+    /// Validate rules in the knowledge base once all files have been loaded
+    pub fn validate_rules(&self) -> PolarResult<()> {
+        let mut kb = self.kb.write().unwrap();
+        // Rewrite namespace implications _before_ validating rule prototypes.
+        kb.rewrite_implications()?;
+        // check rules are valid against rule prototypes
+        kb.validate_rules()
+    }
```
Comment (gj) [inline polar-core/src/polar.rs:244]: Why did you move this into a separate FFI call? What was wrong with calling both methods at the end of `load_source()`?

**Review by gj — CHANGES_REQUESTED** [review]

File: polar-core/src/polar.rs (hunk 1 above)
Comment (leina05) [reply polar-core/src/polar.rs:244]: Since `load_source` is called once per file from the host libraries, if the validation were in there it would get run on every file instead of after all files are loaded. The way I implemented this, the host library is responsible for loading all the files and then calling `validate_rules()`. The alternative would have been adding a way to load multiple files in the core, which I thought would take longer to implement and be a bigger core change.

**Comment by gj** [comment]
superseded by #1138
